addon/components/polaris-resource-list/filter-control.js

```diff
@@ -0,0 +1,328 @@
+import { inject as service } from '@ember/service';
+import { computed, get } from '@ember/object';
+import layout from '../../templates/components/polaris-resource-list/filter-control';
+import { FilterType } from '@smile-io/ember-polaris/components/polaris-resource-list/filter-control/filter-value-selector';
+
+export default Component.extend({
+  tagName: '',
+
+  layout,
+
+  context: service('polaris-resource-list/context'),
+
+  /**
+   * @property searchValue
+   * @type {String}
+   * @default null
+   * @public
+   */
+  searchValue: null,
+
+  /**
+   * @property appliedFilters
+   * @type {Object[]}
+   * @default null
+   * @public
+   */
+  appliedFilters: null,
+
+  /**
+   * @property additionalAction
+   * @type {Object}
+   * @default null
+   * @public
+   */
+  additionalAction: null,
+
+  /**
+   * @property focused
+   * @type {Boolean}
+   * @default false
+   * @public
+   */
+  focused: false,
+
+  /**
+   * @property filters
+   * @type {Object[]}
+   * @default null
+   * @public
+   */
+  filters: null,
+
+  /**
+   * @property onSearchBlur
+   * @type {Function}
+   * @default noop
+   * @public
+   */
+  onSearchBlur() {},
+
+  /**
+   * @property onSearchChange
+   * @type {Function}
+   * @default noop
+   * @public
+   */
+  onSearchChange() {},
+
+  /**
+   * @property onFiltersChange
+   * @type {Function}
+   * @default noop
+   * @public
+   */
+  onFiltersChange() {},
+
+  /**
+   * Button details for `additionalAction`. This is here
+   * instead of in the template because handlebars
+   * always evaluates both branches of an `if`-statement
+   * so will try creating an action from `additionalAction.onAction`
+   * even if it doesn't exist, which leads to an error.
+   *
+   * @property additionalActionButton
+   * @type {Object}
+   * @private
+   */
+  additionalActionButton: computed(
+    'additionalAction.{text,accessibilityLabel,url,external,destructive,icon,loading,onAction}',
+    'context.selectMode',
+    function() {
+      let { additionalAction, context } = this.getProperties(
+        'additionalAction',
+        'context'
+      );
+      if (!additionalAction) {
+        return null;
+      }
+
+      let props = Object.assign({}, additionalAction, {
+        disabled: context.get('selectMode'),
+      });
+
+      // Rename onAction to onClick.
+      props.onClick = props.onAction;
+      delete props.onAction;
+
+      return {
+        componentName: 'polaris-button',
+        props,
+      };
+    }
+  ).readOnly(),
+
+  /**
+   * List of appliedFilters in a format
+   * for rendering in the template
+   *
+   * @property appliedFiltersForRender
+   * @type {Object[]}
+   * @private
+   */
+  appliedFiltersForRender: computed('appliedFilters.[]', function() {
+    let appliedFilters = this.get('appliedFilters') || [];
+    return appliedFilters.map((appliedFilter) => {
+      let appliedFilterForRender = JSON.parse(JSON.stringify(appliedFilter));
```

Comment:
I wonder if there will ever be a nice, normal way to copy objects in JavaScript 🙏

Reply:
We can only hope 🙏 😛